src/Moq/InvocationCollection.cs

```diff
@@ -68,6 +76,9 @@ public void Clear()
 				this.invocations = null;
 				this.count = 0;
 				this.capacity = 0;
+
+				this.owner.Setups.UninvokeAll();
```

Comment:
Drilling into another property of the owning mock isn't particularly nice. It would be nicer to invoke something on the owning mock itself.